packages/plugin-css/src/index.ts

```diff
@@ -286,110 +279,169 @@ export function transformLink(value: ParsedLinkToken['$value']): string {
+      return transformDuration(value);
     }
     case 'font' as 'fontFamily': // @deprecated (but keep support for now)
     case 'fontFamily': {
-      return transformFontFamily(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformFontFamily(value);
     }
     case 'fontWeight': {
-      return transformFontWeight(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformFontWeight(value);
     }
     case 'cubicBezier': {
-      return transformCubicBezier(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformCubicBezier(value);
     }
     case 'number': {
-      return transformNumber(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformNumber(value);
     }
     case 'link': {
-      return transformLink(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformLink(value);
     }
     case 'strokeStyle': {
-      return transformStrokeStyle(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (isAlias(originalVal)) {
+        return varRef(originalVal as string, {prefix: options?.prefix});
+      }
+      return transformStrokeStyle(value);
     }
+    // composite tokens
     case 'border': {
-      return transformBorder(value as typeof token.$value);
+      const {value, originalVal} = getMode(token, options?.mode);
+      if (typeof originalVal === 'string') {
+        return varRef(originalVal, {prefix: options?.prefix});
+      }
+      const width = isAlias(originalVal.width) ? varRef(originalVal.width, {prefix: options?.prefix}) : transformDimension(value.width);
+      const color = isAlias(originalVal.color) ? varRef(originalVal.color, {prefix: options?.prefix}) : transformColor(value.color);
+      const style = isAlias(originalVal.style) ? varRef(originalVal.style, {prefix: options?.prefix}) : transformStrokeStyle(value.style);
+      return `${width} ${style} ${color}`;
     }
     case 'shadow': {
-      return transformShadow(value as typeof token.$value);
+      let {value, originalVal} = getMode(token, options?.mode);
+      if (typeof originalVal === 'string') {
```

Comment:
So a big, annoying change in `plugin-css` was:

**Before**: all tokens would be resolved to CSS variables if possible (i.e. any tokens that were aliases) before they were transformed

**After**: the alias action is moved to each individual token’s transformer

The reason for this change is there’s a lot of nuance and complexity between token values and aliases. Specifically with composite tokens—you can’t really transform the token where some parts may be aliased and some parts may not be, unless you know what token type it is first.

As a result of this PR, the code _feels_ a bit messier. But overall is much more stable than it was before. As evidenced by removing hacks like `CSS_VAR_RE.test(value)` where we didn’t know if something had been transformed or not yet (which we should know; and now we do)

Reply:
Also, as a result of this change, **there are no more transform* methods for composite types.** The reason is that in order to work properly, composite transformers need access to not only the token itself, but which mode is being accessed, and other meta-level information.

So now, the `defaultTransformer()` just keeps all context. Though we can clean up some logic for base tokens, unfortunately, all composite token transformation has to happen in one giant transformer. Otherwise it’s not really separating concerns; you’re just passing a giant context object around between functions which is probably worse

Reply:
Anyways, TL;DR, this also came up as a blocker for #113 as well, but this is the solution (I think). Anyway, we’ll know once I merge this and take another crack at that.